src/survey-config/survey-config.controller.ts

```diff
@@ -0,0 +1,155 @@
+
+@Controller("survey-config")
+@ApiTags("survey-config")
+export class SurveyConfigController {
+  // Create a logger instance for this controller to log events and errors.
+  private readonly logger = new Logger(SurveyConfigController.name);
+
+  // The survey config controller class, injecting the survey config service.
+  constructor(private readonly surveyConfigService: SurveyConfigService) {}
+
+  @Post()
+  @ApiOperation({ summary: "Create a survey config" }) // Describes the api operation for Swagger.
+  @ApiResponse({ status: HttpStatus.CREATED, type: ResponseSurveyConfigDto }) // Describes the response for Swagger.
+  async createSurveyConfig(
+    @Res() res,
+    @Body() createSurveyConfigDto: CreateSurveyConfigDto
+  ) {
+    try {
+      // Log the initiation for the Survey config creation
+      this.logger.log(`Creating a new survey config`);
+      const createdSurveyConfig =
+        await this.surveyConfigService.createSurveyConfig(
+          createSurveyConfigDto
+        );
+
+      // Log the successful creation of the survey config
+      this.logger.log(`Successfully created the new survey config.`);
+
+      // Return a success response with the created request data
+      return res.status(HttpStatus.CREATED).json({
+        message: "Survey config created successfully",
+        data: createdSurveyConfig,
+      });
+    } catch (error) {
+      this.logger.error(`Failed to create new survey config.`, error);
+
+      // Return an error response
+      return res.status(HttpStatus.CREATED).json({
+        message: `Failed to create new survey config.`,
+      });
+    }
+  }
+
+  // Get all survey config
+  @Get()
+  @ApiOperation({ summary: "Get all Survey config" }) // Api operation for swagger
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    type: ResponseSurveyConfigDto,
+    isArray: true,
+  }) // Api response for Swagger.
+  async getAllSurveyConfig(@Res() res, @Query() filter: SurveyConfigFilterDto) {
+    try {
+      this.logger.log(`Initiated fetching all the survey config.`);
+      const surveyConfigs = await this.surveyConfigService.getAllSurveyConfig(
+        filter
+      );
+      this.logger.log(`Successfully fetched survey config`);
+
+      return res.status(HttpStatus.OK).json({
+        message: "Successfully fetched all survey config.",
+        data: surveyConfigs,
+      });
+    } catch (error) {
+      this.logger.error(`Failed to fetch survey configs`, error);
+      return res
+        .status(HttpStatus.INTERNAL_SERVER_ERROR)
+        .json({ message: "Failed to fetch survey configs" });
+    }
+  }
+
+  // Update the survey config
+  @Patch("/update/:id")
+  @ApiOperation({ summary: "Update the survey config by ID" }) // Api operation for swagger
+  @ApiResponse({ status: HttpStatus.OK, type: ResponseSurveyConfigDto }) // Api operation for swagger
+  async updateSurveyConfigById(
+    @Res() res,
+    @Param("id") id: number,
```

Comment:
Add parseInt in params and remove + from all Ids

Reply:
Done